No local image when make docker-build - SyntaxError on make docker-push-dev with python3

/kind bug

**What steps did you take and what happened:**

```bash
make docker-build
```

No error returned but image not listed with `docker images`

```bash
make docker-push-dev
export DEV_DOCKER_REPO=docker.io
export MANAGER_IMAGE_NAME=datalayer/cluster-api-aws-controller
export MANAGER_IMAGE_TAG=latest
make docker-push-dev
```

returns:

```bash
 File "<frozen importlib._bootstrap>", line 626, in _load_backward_compatible
  File "/home/datalayer/.cache/bazel/_bazel_datalayer/991af353125ff5dca82d9f35abab139e/external/puller/file/downloaded/containerregistry/client/docker_creds_.py", line 31, in <module>
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 951, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 894, in _find_spec
  File "<frozen importlib._bootstrap_external>", line 1157, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1131, in _get_spec
  File "<frozen importlib._bootstrap_external>", line 1112, in _legacy_get_spec
  File "<frozen importlib._bootstrap>", line 441, in spec_from_loader
  File "<frozen importlib._bootstrap_external>", line 544, in spec_from_file_location
  File "/home/datalayer/.cache/bazel/_bazel_datalayer/991af353125ff5dca82d9f35abab139e/external/puller/file/downloaded/httplib2/__init__.py", line 942
    print "connect: (%s, %s) ************" % (self.host, self.port)
                                         ^
SyntaxError: invalid syntax
 (/opt/miniconda3/envs/datalayer/bin/python /home/datalayer/.cache/bazel/_bazel_datalayer/991af353125ff5dca82d9f35abab139e/external/puller/file/downloaded --directory /home/datalayer/.cache/bazel/_bazel_datalayer/991af353125ff5dca82d9f35abab139e/external/golang-image/image --name registry.hub.docker.com/library/golang:1.10-alpine)
INFO: Elapsed time: 0.931s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (0 packages loaded)
FAILED: Build did NOT complete successfully (0 packages loaded)
```

Btw no such error with python2.7 but 0 bytes to push...

```bash
INFO: Build completed successfully, 1 total action
docker.io/datalayer/cluster-api-aws-controller:latest was resolved to docker.io/datalayer/cluster-api-aws-controller:latest
docker.io/datalayer/cluster-api-aws-controller:{GIT_VERSION} was resolved to docker.io/datalayer/cluster-api-aws-controller:v1.0.0-alpha.2.27-e74f4f96463735-dirty
E1105 16:27:00.538907    6183 docker_session_.py:332] Error during upload of: docker.io/datalayer/cluster-api-aws-controller:v1.0.0-alpha.2.27-e74f4f96463735-dirty
F1105 16:27:00.539144    6183 fast_pusher_.py:186] Error publishing docker.io/datalayer/cluster-api-aws-controller:v1.0.0-alpha.2.27-e74f4f96463735-dirty: response: {'status': '301', 'connection': 'close', 'content-length': '0', 'location': 'https://www.docker.com/v2/datalayer/cluster-api-aws-controller/blobs/uploads/'}
: None
E1105 16:27:00.553980    6184 docker_session_.py:332] Error during upload of: docker.io/datalayer/cluster-api-aws-controller:latest
F1105 16:27:00.554163    6184 fast_pusher_.py:186] Error publishing docker.io/datalayer/cluster-api-aws-controller:latest: response: {'status': '301', 'connection': 'close', 'content-length': '0', 'location': 'https://www.docker.com/v2/datalayer/cluster-api-aws-controller/blobs/uploads/'}
: None
Makefile:109: recipe for target 'docker-push-dev' failed
```

**What did you expect to happen:**

A local docker image should be build and available on my local env.

**Environment:**

- Cluster-api-provider-aws version:  e74f4f9
- OS (e.g. from `/etc/os-release`): Ubuntu 18.04.1

--- a/Makefile
+++ b/Makefile
@@ -98,17 +98,25 @@ test: generate ## Run tests
 copy-genmocks: test ## Copies generated mocks into the repository
 	cp -Rf bazel-genfiles/pkg/* pkg/
 
+BAZEL_DOCKER_ARGS_COMMON := --define=MANAGER_IMAGE_NAME=$(MANAGER_IMAGE_NAME) --define=MANAGER_IMAGE_TAG=$(MANAGER_IMAGE_TAG) $(BAZEL_ARGS)
+BAZEL_DOCKER_ARGS := --define=DOCKER_REPO=$(STABLE_DOCKER_REPO) $(BAZEL_DOCKER_ARGS_COMMON)
+BAZEL_DOCKER_ARGS_DEV := --define=DOCKER_REPO=$(DEV_DOCKER_REPO) $(BAZEL_DOCKER_ARGS_COMMON)
+
 .PHONY: docker-build
-docker-build: generate ## Build the docker image
-	bazel build //cmd/manager $(BAZEL_ARGS)
+docker-build: generate ## Build the production docker image
+	bazel run //cmd/manager:manager-image $(BAZEL_DOCKER_ARGS)
+
+.PHONY: docker-build-dev
+docker-build-dev: generate ## Build the development docker image
+	bazel run //cmd/manager:manager-image $(BAZEL_DOCKER_ARGS_DEV)
 
 .PHONY: docker-push
 docker-push: generate ## Push production docker image
-	bazel run //cmd/manager:manager-push --define=DOCKER_REPO=$(STABLE_DOCKER_REPO) --define=MANAGER_IMAGE_NAME=$(MANAGER_IMAGE_NAME) --define=MANAGER_IMAGE_TAG=$(MANAGER_IMAGE_TAG) $(BAZEL_ARGS)
+	bazel run //cmd/manager:manager-push $(BAZEL_DOCKER_ARGS)
 
 .PHONY: docker-push-dev
 docker-push-dev: generate ## Push development image
-	bazel run //cmd/manager:manager-push --define=DOCKER_REPO=$(DEV_DOCKER_REPO) --define=MANAGER_IMAGE_NAME=$(MANAGER_IMAGE_NAME) --define=MANAGER_IMAGE_TAG=$(MANAGER_IMAGE_TAG) $(BAZEL_ARGS)
+	bazel run //cmd/manager:manager-push $(BAZEL_DOCKER_ARGS_DEV)
 
 .PHONY: clean
 clean: ## Remove all generated files

--- a/docs/development.md
+++ b/docs/development.md
@@ -8,9 +8,9 @@
 - [Committing code](#committing-code)
 - [Mocks](#mocks)
 - [Set up](#set-up)
-    - [Base requirements](#base-requirements)
-    - [Using Google Cloud](#using-google-cloud)
-        - [Using images on Google Cloud](#using-images-on-google-cloud)
+  - [Base requirements](#base-requirements)
+  - [Using Google Cloud](#using-google-cloud)
+    - [Using images on Google Cloud](#using-images-on-google-cloud)
 - [cluster-api-dev-helper](#cluster-api-dev-helper)
 
 <!-- /TOC -->
@@ -46,7 +46,8 @@ required for the provider manifests.
 3. Install [minikube][minikube]
    - `brew install minikube` on MacOS.
 4. Install [bazel][bazel]
-5. Configure `minikube`:
+5. Configure Python 2.7+ with [pyenv][pyenv] if your default is Python 3.x.
+6. Configure `minikube`:
     1. Use Kubernetes v1.12.1 `minikube config set kubernetes-version v1.12.1`.
     2. Use kubeadm as bootstrapper `minikube config set bootstrapper kubeadm`.
 
@@ -91,3 +92,4 @@ make cluster-api-dev-helper
 [minikube]: https://kubernetes.io/docs/setup/minikube/
 [aws_cli]: https://docs.aws.amazon.com/cli/latest/userguide/installing.html
 [bazel]: https://docs.bazel.build/versions/master/install.html
+[pyenv]: https://github.com/pyenv/pyenv

